cachito/web/content_manifest.py

```diff
@@ -69,12 +75,12 @@ def process_go_package(self, package, dependency):
         """
         Process go-package package.
 
-        :param Package package: the go-package package to process
+        :param ContentManifestPackage package: the go-package package to process
         :param Dependency dependency: the go-package package dependency to process
         """
         if dependency.type == "go-package":
             icm_dependency = {"purl": dependency.to_purl()}
-            self._gopkg_data[package.id]["dependencies"].append(icm_dependency)
+            self._gopkg_data[package.hash]["dependencies"].append(icm_dependency)
```

Comment:
While hash collisions are quite unlikely, I would rather use a key that is guaranteed unique (see next suggestion)

Reply:
Now that package implements `__hash__` and `__eq__`, it should be used as the key directly:
```suggestion
            self._gopkg_data[package]["dependencies"].append(icm_dependency)
```